services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/resource_tracker_service_runs.py

```diff
@@ -58,8 +58,8 @@ async def list_service_runs(
             product_name, user_id, wallet_id, page_params.offset, page_params.limit
         )
     else:
-        msg = "wallet_id and access_all_wallet_usage parameters must be specified together"
-        raise ResourceUsageTrackerCustomRuntimeError(msg=msg)
+        detail = "wallet_id and access_all_wallet_usage parameters must be specified together"
+        raise MyHTTPException(status_code=404, detail=detail)
```

Comment:
`MyHTTPException` looks like a sample code :-)  Please create a more robust logic around your exceptions. What is your intention here?